hazelcast/src/main/java/com/hazelcast/query/impl/predicates/LikePredicate.java

```diff
@@ -43,6 +48,28 @@ public LikePredicate(String attributeName, String expression) {
         this.expression = expression;
     }
 
+    @Override
+    public Set<QueryableEntry> filter(QueryContext queryContext) {
+        Index index = queryContext.getIndex(attributeName);
+        if (indexPrefix == null) {
+            indexPrefix = expression.substring(0, expression.length() - 1);
+        }
+        return index.getRecords(indexPrefix, true, indexPrefix + "\uFFFF", false);
+    }
+
+    @Override
+    public boolean isIndexed(QueryContext queryContext) {
+        Index index = queryContext.getIndex(attributeName);
+        if (index != null && index.isOrdered()) {
+            int firstEntry = expression.indexOf('%');
```

Comment:
As far as I understand, there should be exactly one instance of `%` in the string exactly at its end. There should be no `_` instances anywhere in the string. There is also a possibility of escaping for `%` and `_`, see `Predicates.like` javadoc for more details on the syntax.